### PR TITLE
Command-line driver for MICM

### PR DESCRIPTION
**Pull request by boulderdaze**
Closes [#308](https://github.com/NCAR/micm/issues/308)

Created try-out example configurations with a single grid cell and a set of initial conditions from a CSV file

- chapman
- TS1
- robertson
- carbon bond 5

### CODE REVIEW

**Review by K20shores — CHANGES_REQUESTED** [review]

File: examples/configs/chapman/initial_conditions.csv [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+PHOTO.O3_1,0.366253459937848
+PHOTO.O3_2,0.466253459937848
+EMIS.O1D,0.166253459937848
+EMIS.O,0.266253459937848
+EMIS.O3,0.366253459937848
+LOSS.N2,0.366253459937848
+LOSS.O2,0.566253459937848
+LOSS.CO2,0.166253459937848
+LOSS.Ar,0.966253459937848
+LOSS.H2O,0.866253459937848
```
[the remark this answers is not in the document]
Comment (K20shores) [reply examples/configs/chapman/initial_conditions.csv:24]: never mind, this is probably just a different formulation than the one I know of

File: examples/configs/chapman/initial_conditions.csv [hunk 2]
```diff
@@ -0,0 +1,24 @@
+CONC.M,0.779127760953859
+CONC.Ar,0.963825315875336
+CONC.CO2,0.846014255793546
+CONC.H2O,0.084410107675724
+CONC.N2,0.300322986940341
+CONC.O1D,0.925827943758015
+CONC.O,0.566253459937848
+CONC.O2,0.566253459937848
+CONC.O3,0.566253459937848
+ENV.temperature,287.45
```
Comment (dwfncar) [inline examples/configs/chapman/initial_conditions.csv:10]: Do we want to use environmental conditions for the stratosphere, say around z = 30 km?

File: examples/example.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    state.PrintHeader();
+    state.PrintState(0);
+
+    // Depending on how stiff the system is
+    // the solver integration step may not be able to solve for the full time step
+    // so we need to track how much time the solver was able to integrate for and continue
+    // solving until we finish
+    double elapsed_solve_time = 0;
+
+    while (elapsed_solve_time < time_step)
```
Comment (dwfncar) [inline examples/example.cpp:174]: Perhaps also an option for multiple time steps would be nice at some point.

**Review by dwfncar — APPROVED** [review]

File: examples/configs/chapman/initial_conditions.csv (hunk 2 above)
Comment (K20shores) [reply examples/configs/chapman/initial_conditions.csv:10]: that would make slightly more sense for chapman

**Review by K20shores — APPROVED** [review]

**Comment by K20shores** [comment]
@boulderdaze would you please add support for running carbon bond 5? There are conditions for the rate constants [all the way back at this commit](https://github.com/NCAR/micm/blob/d67ce1bc8acacc28495ed8e9ff5f0a4d1f21f14c/test/tutorial/test_jit_tutorial.cpp)

**Comment by boulderdaze** [comment]
> @boulderdaze would you please add support for running carbon bond 5? There are conditions for the rate constants [all the way back at this commit](https://github.com/NCAR/micm/blob/d67ce1bc8acacc28495ed8e9ff5f0a4d1f21f14c/test/tutorial/test_jit_tutorial.cpp)

Yes, `carbon bond 5` configuration is added

File: include/micm/solver/state.inl [hunk 4]
```diff
@@ -105,11 +105,75 @@ namespace micm
...
+
+    // Print only the first 10 if the number of elements are bigger than 10
+    if (variable_names_.size() > 10)
+    {
+      for(short i=0; i<10; i++)
+      {
+        std::cout << "," << std::setw(width) << variable_names_[i];
+      }
+      std::cout << std::setw(width) << "..." << std::endl;
+    }
```
Comment (K20shores) [inline include/micm/solver/state.inl:134]: We should print everything. That way the output could be redirected to a file.

File: include/micm/solver/state.inl [hunk 5]
```diff
@@ -105,11 +105,75 @@ namespace micm
...
+
+    // Print only the first 10 if the number of elements are bigger than 10
+    if (variable_names_.size() > 10)
+    {
+      for(short i=0; i<10; i++)
+      {
+        std::cout << std::scientific << "," << std::setw(width) << std::setprecision(2) << variables_[0][variable_map_[variable_names_[i]]];
+      }
+      std::cout << std::scientific << std::setw(width) << std::setprecision(2) << "...";
+    }
```
Comment (K20shores) [inline include/micm/solver/state.inl:167]: Same here, please print everything

**Review by K20shores — CHANGES_REQUESTED** [review]

**Review by mattldawson — APPROVED** [review]
looks great!

**Pull request merged**